packages/realm/src/indirect.ts

```diff
@@ -47,9 +47,14 @@ type Indirects = {
  */
 export const indirect = {} as Indirects;
 
+const IGNORED_PROPS = new Set([
+  // See https://github.com/realm/realm-js/issues/6522
+  "$$typeof",
```

Comment:
This was causing me issues before and existed pre-refactor; was the new refactor meant to resolve this? @kraenhansen

Reply:
I didn't think of that issue 👍 good catch.